cips/cip-031.md

```diff
@@ -122,23 +156,59 @@ When upgrading to v4 we propose introducing a migration that will set the Parame
 
 # Backwards Compatibility
 
-All existing lockup accounts will be subject to the locking system. This change will take effect in v4, since this is a larger upgrade there is not a need for backwards compatible systems.
+Continuous Vesting, Delayed Vesting will implement the `UpdateSchedule` function allowing for the original amount to be added to. Periodic and Permanent will implement the design as a noop, and they will be disallowed from being created. 
```

Comment:
```suggestion
The continuous and delayed vesting account types will be updated to implement CIP-31. Specifically, they will implement the `UpdateSchedule` function so that the original vesting amount can account for staking rewards. The periodic and permanent account types will not support CIP-31. They will be disallowed from being created. 
```

[question] why do we need to disallow creating them? What about periodic and permanent vesting accounts that already exist?

Reply:
disallowing creation of them in order to simplify the protocol and due to proposing not implementing update schedule. Existing accounts, of which I couldnt find any, would still work, only the creation is disallowed

Reply:
If existing accounts still work then I don't understand why celestia-app can't support creating new ones. 

Proposal: 
- Implement no-op `UpdateSchedule` for permanent and vesting accounts
- Preserve the ability to create new permanent and vesting accounts
- Clarify CIP-31 only applies to continuous and delayed vesting accounts

Reply:
Also given the most recent iteration of the implementation takes all staking rewards and updates the original vesting amount, it seems feasible to implement CIP-31 for periodic vesting accounts. AFAIU the complexity in the first implementation occurred when updating the amount to vest for various periods but that logic no longer needs to be added, right?

Reply:
for periodic this doesnt work as periods are proposed in json on creation. You need to update the periods as the periods contain the amount unlocked.

Reply:
Got it, thanks! Resolving.